src/main/java/com/bobocode/svydovets/bibernate/session/SessionImpl.java

```diff
@@ -111,9 +118,40 @@ public void close() {
     }
 
     @Override
-    public <T> T merge(T entity) {
+    public <T> T merge(T detachedEntity) {
```

Comment:
I guess param name should be just entity, because technically we can pass entity with any state,
and merge function responsible for manage the logic
If we will pass only entities with detached state we no longer need to check for state or throw an exception